### PR TITLE
Bring back algebraName at Instrumentation and add it to Aspect

### PR DESCRIPTION
**Pull request by Fristi**
It's quite convenient to have the `algebraName` at the site when transforming the algebra:

```scala
val woven = new PostgresRepos[Aspect.Weave.Codomain[RIO[OpenTracing, *], TraceData, *]] {
   // when call `Derive.aspect.algebraName` you can access it's name, but it doesn't make sense here yet
   def depots = Derive.aspect.weave(repos.depots)
  // more repos here
}

val traceTransformation = new ((Aspect.Weave.Codomain[RIO[OpenTracing, *], TraceData, *]) ~> RIO[OpenTracing, *]) {
  def apply[A](fa: Codomain[RIO[OpenTracing, *], TraceData, A]): RIO[OpenTracing, A] = {
    fa.codomain.target.flatMap { res =>
      fa.codomain.instance.extract(res) match {
        case Some(value) =>
          // here I also would like to access algebra name, to add it to the tag
          OpenTracing.tag("result", value) *> ZIO.succeed(res)
        case None => ZIO.succeed(res)
      }
    }.span(fa.codomain.name)
  }
}

woven.mapK(traceTransformation)
```

### CODE REVIEW

**Comment by joroKr21** [comment]
I think it should be part of `Weave`, not of `Advice`.

**Comment by Fristi** [comment]
Yea good point! Should the method name also be in there ?

**Comment by joroKr21** [comment]
`Advice.name` the method name in `codomain` and parameter name in `domain`.

**Comment by Fristi** [comment]
Makes sense! Fixed it

File: core/src/main/scala/cats/tagless/aop/Aspect.scala [hunk 1]
```diff
@@ -86,10 +87,10 @@ object Aspect {
   object Advice {
     type Aux[F[_], G[_], T] = Advice[F, G] { type A = T }
 
-    def apply[F[_], G[_], T](adviceName: String, adviceTarget: F[T])(implicit G: G[T]): Aux[F, G, T] =
+    def apply[F[_], G[_], T](method: String, adviceTarget: F[T])(implicit G: G[T]): Aux[F, G, T] =
```
Comment (joroKr21) [inline core/src/main/scala/cats/tagless/aop/Aspect.scala:90]: Let's revert this change because for `domain` it's the argument's name
```suggestion
    def apply[F[_], G[_], T](adviceName: String, adviceTarget: F[T])(implicit G: G[T]): Aux[F, G, T] =
```

File: core/src/main/scala/cats/tagless/aop/Aspect.scala [hunk 2]
```diff
@@ -86,10 +87,10 @@ object Aspect {
...
       new Advice[F, G] {
         type A = T
-        val name = adviceName
+        val name = method
```
Comment (joroKr21) [inline core/src/main/scala/cats/tagless/aop/Aspect.scala:93]:
```suggestion
        val name = adviceName
```
